pacsman/pynetdicom_client.py

```diff
@@ -0,0 +1,399 @@
+                    study_ids.append(result.StudyInstanceUID)
+
+            return study_ids
+
+    def series_for_study(self, study_id, modality_filter=None):
+        """
+        :param study_id: StudyInstanceUID from PACS
+        :return: SeriesInfo
+        """
+        ae = AE(ae_title=self.client_ae, scu_sop_class=QueryRetrieveSOPClassList)
+
+        with association(ae, self.pacs_url, self.pacs_port) as assoc:
+            dataset = Dataset()
+            dataset.StudyInstanceUID = study_id
+
+            # Filtering modality with 'MR\\CT' doesn't seem to work with pynetdicom
+            dataset.Modality = ''
+            dataset.PatientName = ''
+            dataset.BodyPartExamined = ''
+            dataset.SeriesDescription = ''
+            dataset.SeriesDate = ''
+            dataset.SeriesTime = ''
+            dataset.SeriesInstanceUID = ''
+            dataset.PatientPosition = ''
+            dataset.QueryRetrieveLevel = 'SERIES'
+
+            responses = assoc.send_c_find(dataset, query_model='S')
+
+            series_infos = []
+            for (status, series) in responses:
+                logger.debug(status)
+                logger.debug(series)
+
+                if status.Status not in status_success_or_pending:
+                    raise Exception('Series C-FIND Failure Response: 0x{0:04x}'.format(status.Status))
+
+                if series and (modality_filter is None or
+                               getattr(series, 'Modality', '') in modality_filter):
+                    description = getattr(series, 'SeriesDescription', '')
+                    body_part_examined = getattr(series, 'BodyPartExamined', None)
+                    if body_part_examined:
+                        description += f' ({body_part_examined})'
+
+                    with association(ae, self.pacs_url, self.pacs_port) as series_assoc:
+                        series_dataset = Dataset()
+                        series_dataset.SeriesInstanceUID = series.SeriesInstanceUID
+                        series_dataset.QueryRetrieveLevel = 'IMAGE'
+                        series_dataset.SOPInstanceUID = ''
+
+                        series_responses = series_assoc.send_c_find(series_dataset, query_model='S')
+                        image_ids = []
+                        for (instance_status, instance) in series_responses:
+                            logger.debug(instance)
+                            if instance_status.Status in status_success_or_pending:
+                                if hasattr(instance, 'SOPInstanceUID'):
+                                    image_ids.append(instance.SOPInstanceUID)
+                            else:
+                                raise Exception(
+                                    'Image C-FIND Failure Response: 0x{0:04x}'.format(
+                                        status.Status))
+
+                    info = SeriesInfo(series_id=series.SeriesInstanceUID, description=description,
```

Comment:
Why not return the entire DICOM dataset, instead of just a named tuple with just a few bits of data from the dataset?

@yujanshrestha what are your thoughts on this?

Pros of returning the named tuple:

- Small amount of data is returned
- Can handle some logic on behalf of our users

Cons of returning the named tuple:
- User won't have access to other information on the dataset
- Our library may be too tightly coupled to the needs of our first applications

Reply:
I mostly set these up like this to be independent of the dicom client library used. For use with `dcmtk`, the raw results would just be some `stdout` lines that would need to be parsed (with no Python-native data type). But like you mentioned, just having that small subset of fields means we have to fiddle with the namedtuple and several usages if a field needs to be added later on for another application. It would probably be more useful to just return a generic bucket of attributes or even just a `dict` with fields like `Modality` or `SOPInstanceUID` as keys.